test/Microsoft.NET.Build.Tests/AppHostTests.cs

```diff
@@ -319,6 +319,40 @@ public void It_can_disable_cetcompat(bool? cetCompat)
             isCetCompatible.Should().Be(!cetCompat.HasValue || cetCompat.Value);
         }
 
+        [Fact]
+        public void It_does_not_configure_dotnet_search_options_on_build()
+        {
+            var targetFramework = ToolsetInfo.CurrentTargetFramework;
+            var runtimeIdentifier = EnvironmentInfo.GetCompatibleRid(targetFramework);
+
+            var testProject = new TestProject()
+            {
+                Name = "AppHostDotNetSearch",
+                TargetFrameworks = targetFramework,
+                RuntimeIdentifier = runtimeIdentifier,
+                IsExe = true,
+            };
+            testProject.AdditionalProperties.Add("AppHostDotNetSearch", "AppRelative");
+            testProject.AdditionalProperties.Add("AppHostRelativeDotNet", "subdirectory");
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute()
+                .Should()
+                .Pass();
+
+            // Output apphost should not have .NET search location options changed, so it
+            // should run successfully with the DOTNET_ROOT environment variable set
+            var outputDirectory = buildCommand.GetOutputDirectory(runtimeIdentifier: runtimeIdentifier);
```

Comment:
Would this otherwise fail because the `subdirectory` folder doesn't exist or doesn't contain a .NET runtime?